### ISSUE TITLE
IDE0031 doesn't account for "is not null"

### ISSUE THREAD

**Issue by v-kydela** [dotnet/roslyn#76093]
**Version Used**: current

**Steps to Reproduce**:

1. Use the ternary operator in a way that could be simplified using null propagation
2. Modify the conditional to use the `is not null` pattern
3. Observe that the analyzer no longer recommends simplifying the expression

```c#
string? str = null;
        
int? a = str == null ? null : str.Length; // Triggers null propagation suggestion
int? b = str is null ? null : str.Length; // Triggers null propagation suggestion
int? c = str != null ? str.Length : null; // Triggers null propagation suggestion
int? d = str is not null ? str.Length : null; // Doesn't trigger null propagation suggestion
```

**Diagnostic Id**:

[IDE0031: Null check can be simplified](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0031)

**Expected Behavior**:

Null propagation should be suggested in all four cases

**Actual Behavior**:

Null propagation is suggested in every case except for the one using `is not null`

**Comment by CyrusNajmabadi** [comment]
This doesn't repro for me:

![Image](https://github.com/user-attachments/assets/cfac4d42-af7f-4063-8591-06a1563a93cf)
